lib/Driver/Driver.cpp

```diff
@@ -2031,6 +2031,12 @@ const ToolChain *Driver::getToolChain(const ArgList &Args) const {
       TC = new toolchains::Darwin(*this, Target);
       break;
     case llvm::Triple::Linux:
+      if (Target.isAndroid()) {
```

Comment:
@compnerd What was the problem in Clang with a hierarchy of toolchains? It seems to me that littering branches everywhere creates very fiddly code, whereas a hierarchy with proper extension points makes it clear which parts might be different.